### PR TITLE
Add regression test workflow

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- Add a regression test workflow which is based on MLFlow Tracking server for better maintenance and automated test workflow.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: for_developers/regression_test/README.md
Comment (vinnamkim) [inline for_developers/regression_test/README.md:1]: Please see this README.md file carefully to see the overview of regression test workflow.

File: tests/regression/test_regression.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    f"{key}={value}"
+                    for key, value in test_case.dataset.extra_overrides.items()
+                ]
+                _, metrics = otx_train(overrides)
+
+                # Submit metrics to MLFlow Tracker server
+                mlflow.log_metrics(metrics)
+
+
+class TestMultiClassCls(BaseTest):
```
Comment (vinnamkim) [inline tests/regression/test_regression.py:87]: @kprokofi, @eugene123tw, @sungmanc, @jaegukhyun 

I'd like to request you guys to read this carefully and horizontally scale this test code for tasks you already worked after this PR.

File: pyproject.toml [hunk 2]
```diff
@@ -237,6 +239,10 @@ exclude = [
     "src/otx/core/engine/utils/*",
     "src/otx/algo/classification/backbones/*",
     "for_developers/helpers.py",
+
+    # Ruff complains it but don't know how to fix since it literally showed no useful logs.
+    # https://github.com/openvinotoolkit/training_extensions/actions/runs/7176557723/job/19541622452?pr=2718#step:5:170
```
Comment (vinnamkim) [inline pyproject.toml:244]: Please let me know if someone has an idea to fix this.

Comment (sungmanc) [reply pyproject.toml:244]: It seems that ruff already fixed errors, so you just save the modified folder and `add --> commit --> push` might resolve the error, I think.

**Review by sungmanc — DISMISSED** [review]
Thanks for the nice introduction. BTW, is there no SDL issue to use the postgress DB? maybe we need confirm from Yunchu?

File: pyproject.toml (hunk 2 above)
Comment (vinnamkim) [reply pyproject.toml:244]: See my local has no problem on https://github.com/openvinotoolkit/training_extensions/pull/2718/commits/676bdbd230a5a69a3b0c82018e8c3ac9dcd37ccf
image

Comment (vinnamkim) [reply pyproject.toml:244]: However, CI's pre-commit raised an error but does not give any useful information: https://github.com/openvinotoolkit/training_extensions/actions/runs/7177128017/job/19543146905?pr=2718#step:5:174
image

**Comment by vinnamkim** [comment]
> Thanks for the nice introduction. BTW, is there no SDL issue to use the postgress DB? maybe we need confirm from Yunchu?

Thanks, I'll contact him.

File: pyproject.toml (hunk 2 above)
Comment (sungmanc) [reply pyproject.toml:244]: Hmm, it seems weird. In my experience, when ruff modified my files, he said to me like below.
image
So, the only thing I need to do is add the modified files and then re-push. 

So, I'm also not sure the reason of the current issue, let's keep it as ignoring.

**Review by jaegukhyun — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]
LGTM!

**Pull request merged**